job-service/src/main/java/com/hpe/caf/services/job/api/JobsResume.java

```diff
@@ -15,55 +15,134 @@
+            if (jobStatus == Job.StatusEnum.PAUSED) {
+                final String resumeJobQueue = config.getResumeJobQueue();
+                try {
+                    // Send a message to the resume job queue.
+                    LOG.debug("Sending a message to the {} queue...", resumeJobQueue);
+                    sendResumeJobMessage(partitionId, jobId, resumeJobQueue, config);
+                } catch (final Exception messagingException) {
+                    final String errorMessage = String.format("Failed to send a message to the %s queue.", resumeJobQueue);
+                    throw new Exception(errorMessage, messagingException);
+                }
+                //  Update the status of the job in the database.
+                LOG.debug("Updating the status of the job in the database...");
+                databaseHelper.resumeJob(partitionId, jobId);
+            }
+
+            LOG.debug("Done.");
         } catch (Exception e) {
-            LOG.error("Error - '{}'", e.toString());
+            LOG.error("Error - '{}'", e.toString(), e);
             throw e;
         }
     }
+
+    private static void sendResumeJobMessage(
+        final String partitionId,
+        final String jobId,
+        final String resumeJobQueue,
+        final AppConfig config)
+        throws IOException, TimeoutException, InterruptedException, ModuleLoaderException, Exception
+    {
+        final Codec codec = ModuleLoader.getService(Codec.class);
+        try (final QueueServices queueServices = QueueServicesFactory.create(config, resumeJobQueue, codec)) {
+            final WorkerAction resumeJobWorkerAction = createResumeJobWorkerAction(partitionId, jobId, resumeJobQueue);
+            queueServices.sendMessage(partitionId, jobId, resumeJobWorkerAction, config, false);
+        }
+    }
+
+    private static WorkerAction createResumeJobWorkerAction(
+        final String partitionId,
+        final String jobId,
+        final String resumeJobQueue) throws CodecException
+    {
+        final WorkerAction workerAction = new WorkerAction();
+        workerAction.setTaskClassifier(DocumentWorkerConstants.DOCUMENT_TASK_NAME);
+        workerAction.setTaskApiVersion(1);
+        workerAction.setTaskData(createResumeJobTaskData(partitionId, jobId));
+        workerAction.setTaskPipe(resumeJobQueue);
+        return workerAction;
+    }
+
+    private static Map<String, byte[]> createResumeJobTaskData(
+        final String partitionId,
+        final String jobId) throws CodecException
+    {
+        final DocumentWorkerDocumentTask documentWorkerDocumentTask = new DocumentWorkerDocumentTask();
+        final Map<String, String> customData = new HashMap<>();
+        customData.put("partitionId", partitionId);
+        customData.put("jobId", jobId);
+        documentWorkerDocumentTask.customData = customData;
+        return OBJECT_MAPPER.convertValue(documentWorkerDocumentTask, Map.class);
```

Comment:
I'm still not comfortable with this line:

![image](https://user-images.githubusercontent.com/20627297/115738878-5e1f0080-a385-11eb-8a89-6f3ccb723a8c.png)

Reply:
I tried using a `TypeReference` here instead of a `Map`:

```
import com.fasterxml.jackson.core.type.TypeReference;
...
private static final TypeReference<Map<String, byte[]>> TASK_DATA_TYPE_REFERENCE
        = new TypeReference<Map<String, byte[]>>()
    {
    };
...
return OBJECT_MAPPER.convertValue(documentWorkerDocumentTask, TASK_DATA_TYPE_REFERENCE);
`

but seeing some errors, I'll have to investigate more:

java.lang.Exception: Failed to send a message to the testResumeJobQueue queue.
	at com.hpe.caf.services.job.api.JobsResumeTest.testResumeJob_Success(JobsResumeTest.java:86)
Caused by: java.lang.IllegalArgumentException: 
Cannot deserialize instance of `[B` out of START_OBJECT token
 at [Source: UNKNOWN; line: -1, column: -1] (through reference chain: java.util.LinkedHashMap["customData"])
	at com.hpe.caf.services.job.api.JobsResumeTest.testResumeJob_Success(JobsResumeTest.java:86)

Reply:
I don't get those warnings in my IDE (I use netbeans), but I've changed this so that I'm using the object mapper to write to bytes, then as a string, the same way the`DatabasePoller` does it:

https://github.com/JobService/job-service/blob/develop/job-service-scheduled-executor/src/main/java/com/hpe/caf/services/job/scheduled/executor/DatabasePoller.java#L72

workerAction.setTaskData(new String(jtd.getTaskData(), StandardCharsets.UTF_8));

Reply:
> I don't get those warnings in my IDE

Your branch would need to be using the latest `caf-common-parent`, that includes the [a5da1ff](https://github.com/CAFapi/caf-common-parent/commit/a5da1ffe44ff3754a778473b3fd0dd27408ea803) commit, to see them.